tensorflow/core/kernels/depthtospace_op.cc

```diff
@@ -112,7 +112,7 @@ class DepthToSpaceOp : public OpKernel {
     auto Tinput = input.tensor<T, kDims>();
     auto Toutput = outputs_tensor->tensor<T, kDims>();
 
-    if (std::is_same<Device, GPUDevice>::value) {
+    if constexpr (std::is_same_v<Device, GPUDevice>) {
```

Comment:
We only use C++14 for the time being. Is there another way?

Reply:
1. Force people to build GPUDevice interface/librarie if debug mode
2. Contact MSVC to force the compile time `if` evaluation even without constexpr (I dont have a knowledge of a flag that does that)

Reply:
@chsigg Can you please take a look on above comments from @Niram7777. Thanks!

Reply:
I think a template specialization should work here.

Reply:
@Niram7777 Can you please check @chsigg's comments and keep us posted. Thanks!

Reply:
Hi, sorry for delay, my company migrated to TensorflowLite and I was quite busy on personal stuffs.
I will submit a MR hopefully this weekend.
I will try the template specialization on SpaceToDepthOp in case of GPUDevice not defined but I hope it wont be too much of pain with c++14

Reply:
> I think a template specialization should work here.

Indeed, a template specialization does the trick -- see PR #42676 (I at the same time also worked on this issue and my PR just got merged into master).